### PR TITLE
added arg and computing state in function of some overrides

### PR DESCRIPTION
**Pull request by loocapro**
## Motivation
Related to #6980

## Solution
Added overrides arg and taking into account when computing the call.

### CODE REVIEW

File: crates/anvil/src/eth/api.rs [hunk 1]
```diff
@@ -2159,8 +2177,13 @@ impl EthApi {
         }
 
         self.backend
-            .with_database_at(Some(block_request), |state, block| {
-                self.do_estimate_gas_with_state(request, state, block)
+            .with_database_at(Some(block_request), |state, block| match overrides {
+                None => self.do_estimate_gas_with_state(request, state, block),
+                Some(overrides) => {
+                    let state =
+                        state::apply_state_override(overrides.into_iter().collect(), state)?;
```
Comment (mattsse) [inline crates/anvil/src/eth/api.rs:2184]: this can be simplified with an if let Some instead

File: crates/anvil/src/eth/api.rs [hunk 2]
```diff
@@ -2144,12 +2156,18 @@ impl EthApi {
...
     ) -> Result<U256> {
         let block_request = self.block_request(block_number).await?;
         // check if the number predates the fork, if in fork mode
         if let BlockRequest::Number(number) = block_request {
             if let Some(fork) = self.get_fork() {
                 if fork.predates_fork(number) {
+                    if overrides.is_some() {
+                        return Err(BlockchainError::StateOverrideError(
+                            "not available on past forked blocks".to_string(),
+                        ));
```
Comment (mattsse) [inline crates/anvil/src/eth/api.rs:2169]: this might be possible with future api, but for now this is ok

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic style nit

File: crates/anvil/tests/it/transaction.rs [hunk 3]
```diff
@@ -961,7 +964,51 @@ async fn estimates_gas_on_pending_by_default() {
...
+    assert!(error_result.is_err(), "Expected an error due to insufficient funds");
+    let error_message = error_result.unwrap_err().to_string();
+    assert!(
+        error_message.contains("Insufficient funds for gas * price + value"),
+        "Error message did not match expected: {}",
+        error_message
+    );
+
+    // Setup state override to simulate sufficient funds for the recipient.
+    let recipient_address_formatted =
```
Comment (loocapro) [inline crates/anvil/tests/it/transaction.rs:993]: Is there a better way to go from H160 to Alloy::Address? @mattsse

**Review by mattsse — APPROVED** [review]
nice!

**Pull request merged**